### PR TITLE
feat(rpc-types-mev): add mev-share sse types

### LINKED ISSUE #1417: Move `mev-share-sse` types to alloy mev crate
### Component

rpc

### Describe the feature you would like

Move https://github.com/paradigmxyz/mev-share-rs/blob/main/crates/mev-share-sse/src/types.rs into alloy mev crate

ref: https://github.com/paradigmxyz/mev-share-rs/issues/44#issuecomment-2388577243

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SkandaBhat**
## Motivation

Add mev-share-sse types from https://github.com/paradigmxyz/mev-share-rs/blob/main/crates/mev-share-sse/src/types.rs as referenced in this [comment](https://github.com/paradigmxyz/mev-share-rs/issues/44#issuecomment-2388577243)

fixes #1417 

## Solution

Ported the types into sse.rs. 

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rpc-types-mev/src/lib.rs [hunk 1]
```diff
@@ -12,6 +12,9 @@ pub use eth_calls::*;
 mod mev_calls;
 pub use mev_calls::*;
 
+mod sse;
+pub use sse::*;
```
Comment (mattsse) [inline crates/rpc-types-mev/src/lib.rs:16]: I'd rather have this as pub mod and not export on global

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit

**Comment by mattsse** [comment]
also please use `cargo +nightly fmt`

File: crates/rpc-types-mev/src/lib.rs [hunk 2]
```diff
@@ -12,6 +12,8 @@ pub use eth_calls::*;
 mod mev_calls;
 pub use mev_calls::*;
 
+pub mod sse;
```
Comment (mattsse) [inline crates/rpc-types-mev/src/lib.rs:15]: oh actually can we rename this to `mevshare`

**Review by mattsse — CHANGES_REQUESTED** [review]
naming nit

File: crates/rpc-types-mev/src/lib.rs (hunk 2 above)
Comment (SkandaBhat) [reply crates/rpc-types-mev/src/lib.rs:15]: mevshare_sse or just mevshare?

Comment (SkandaBhat) [reply crates/rpc-types-mev/src/lib.rs:15]: or maybe mevshare_events? Because the types are associated with server sent events for mevshare

Comment (mattsse) [reply crates/rpc-types-mev/src/lib.rs:15]: just mevshare

Comment (SkandaBhat) [reply crates/rpc-types-mev/src/lib.rs]: done in 224edab61762bdc3566817a5a62fe77f42a93e0e

**Pull request merged**